vividus-plugin-saucelabs/src/main/java/org/vividus/selenium/sauce/SauceConnectOptions.java

```diff
@@ -98,8 +103,25 @@ public String build(String tunnelIdentifier) throws IOException
 
     private Path createPacFile(String tunnelIdentifier) throws IOException
     {
+        List<String> skipProxyHosts = getSkipProxyHosts();
+        skipProxyHosts.addAll(SAUCELABS_HOSTS);
         return createTempFile("pac-saucelabs-" + tunnelIdentifier, ".js",
-                String.format(PAC_FILE_CONTENT_FORMAT, skipProxyHostsPattern, getProxy()));
+                String.format(PAC_FILE_CONTENT_FORMAT, getSkipShExpMatcher(skipProxyHosts), getProxy()));
+    }
+
+    private String getSkipShExpMatcher(List<String> skipProxyHosts)
+    {
+        return skipProxyHosts.stream()
+                             .map(host -> String.format("shExpMatch(host, \"%s\")", host))
+                             .collect(Collectors.joining(" || "));
+    }
+
+    private List<String> getSkipProxyHosts()
+    {
+        return Optional.ofNullable(split(skipHostGlobPatterns, ','))
```

Comment:
Can Spring do parsing for us?

Reply:
if the value is not set we will get null collection imho code looks better without collection nulls checks

Reply:
You may be able to resolve this via SPEL

Reply:
this might be broken by end users since they do not know SPEL

Reply:
discussed to use null collection check for now but in the future to add special property editor for collections -> https://github.com/vividus-framework/vividus/issues/2621